### PR TITLE
feat(snackbar): Auto Hide for Snackbar

### PR DESCRIPTION
**Pull request by kacepe**
I have created auto hide for snackbar.
`autoHideDuration: number | boolean` - The number of milliseconds to wait before automatically dismissing. If no value is specified the snackbar will dismiss normally.
- Added `autoHideDuration` option to `MdSnackBarConfig`
- Snackbar hide after `autoHideDuration`  If no value is specified the snackbar will dismiss normally.
- If a snackbar is dismissed before the timer expires, the timer will be cleared.

r: @josephperrott

### CODE REVIEW

**Comment by kacepe** [comment]
I signed it!

**Comment by kacepe** [comment]
I signed it!

**Comment by kacepe** [comment]
r: @jelbourn

File: src/lib/snack-bar/snack-bar-ref.ts [hunk 1]
```diff
@@ -19,6 +19,8 @@ export class MdSnackBarRef<T> {
   /** Subject for notifying the user that the snack bar has closed. */
   private _afterClosed: Subject<any> = new Subject();
 
+  autoHide: number;
```
Comment (jelbourn) [inline src/lib/snack-bar/snack-bar-ref.ts]: `_autoDismissTimeoutId`

File: src/lib/snack-bar/snack-bar-ref.ts [hunk 2]
```diff
@@ -36,6 +38,10 @@ export class MdSnackBarRef<T> {
         this._afterClosed.complete();
       });
     }
+    // Clear autohide interval
+    if (this.autoHide) {
+      clearInterval(this.autoHide);
```
Comment (jelbourn) [inline src/lib/snack-bar/snack-bar-ref.ts]: If you use `setTimeout`, the corresponding cancel function is `cancelTimeout`

File: src/lib/snack-bar/snack-bar.ts [hunk 3]
```diff
@@ -56,6 +54,12 @@ export class MdSnackBar {
     } else {
       mdSnackBarRef.containerInstance.enter();
     }
+    // Auto dismiss after timeout.
```
Comment (jelbourn) [inline src/lib/snack-bar/snack-bar.ts]: Comment like
```ts
// If the snackbar has a specified duration, set up a timeout to automatically dismiss it.
```

File: src/lib/snack-bar/snack-bar.ts [hunk 4]
```diff
@@ -56,6 +54,12 @@ export class MdSnackBar {
...
+    if (config.autoHideDuration) {
+      mdSnackBarRef.autoHide = setTimeout(() => {
```
Comment (jelbourn) [inline src/lib/snack-bar/snack-bar.ts]: You don't need the braces:
```ts
mdSnackBarRef._autoDismissTimeoutId = 
    setTimeout(() => mdSnackBarRef.dismiss(), config.duration);
```

**Comment by jelbourn** [comment]
Closing in favor of #1856

**Pull request closed without merging**